app/serializers/page_serializer.rb

```diff
@@ -32,6 +33,13 @@ def latest_blog_post_links
     end
   end
 
+  def related_links
+    object.related_links(3).map do |article|
+      article = article.mirrors.first if current_locale == 'cy'
+      build_json_link article
+    end
+  end
+
```

Comment:
The private build_json_link method, refers to page more than self. I think we can remove this method and replaces and move the code to another class. Maybe could be PageLinkSerializer with title and path as attribute.